File: Maximum Product of Word Lengths.java
class Solution {
    public int maxProduct(String[] words) 
    {
    if(words==null || words.length==0)
        return 0;
 
    int[] arr = new int[words.length];    
    for(int i=0; i<words.length; i++){
        for(int j=0; j<words[i].length(); j++){
            char c = words[i].charAt(j);
            arr[i] |= (1<< (c-'a'));
        }
    }    
 
    int result = 0;
 
    for(int i=0; i<words.length; i++)
    {
        for(int j=i+1; j<words.length; j++)
        {
            if((arr[i] & arr[j]) == 0)
                result = Math.max(result, words[i].length()*words[j].length());
        }
    }
    return result;
    }
}

File: Maximum Product of Word Lengths.cpp
class Solution {
public:
    int maxProduct(vector<string>& words) 
    {
        int n=words.size();
        vector<int> res(n);
        int ans = 0;
        for(int i = 0; i < n; i++)
        {
            for(char c : words[i])
                res[i] |= (1 << (c-'a'));
            cout<<res[i]<<endl;
        }
        for(int i = 0; i < n; i++)
        {
            auto it = res.begin()+i;
            while(it != res.end())
            {
                   it = find_if(it+1, res.end(), [&res, i](const int& e){
                       return ((res[i] & e) == 0);
                   });
                if(it != res.end()){
                    ans = max(ans, (int)words[i].size()*(int)words[it-res.begin()].size());
                }
            }
            
        }
        return ans;
    }
};
